File: JAVA/Piece.java

public abstract class Piece {
    private boolean isKilled = false;
    private boolean isWhitePiece = true;

    public Piece(boolean isWhitePiece) {
        this.isWhitePiece = isWhitePiece;
    }

    public boolean isWhite() {
        return isWhitePiece;
    }

    public void setWhite(boolean isWhitePiece) {
        this.isWhitePiece = isWhitePiece;
    }

    public boolean isKilled() {
        return isKilled;
    }

    public void setKilled(boolean isKilled) {
        this.isKilled = isKilled;
    }

    public abstract boolean canMove(Board board, Spot start, Spot end);
    public abstract String getSymbol();
    public abstract void move(Board board, Spot start, Spot end);
}

File: piece.cpp
#include "piece.hpp"
#include "spot.hpp"

Piece::Piece(bool white) : isWhitePiece(white) {
    isPieceKilled = false;
}

bool Piece::isWhite() {
    return isWhitePiece;
}

void Piece::setWhite(bool pIswhite) {
    isWhitePiece = pIswhite;
}

bool Piece::isKilled() {
    return isPieceKilled;
}

void Piece::setKilled(bool killed) {
    this->isPieceKilled = killed;
}
